test/extended/marketplace/source_create.go

```diff
@@ -0,0 +1,101 @@
+package marketplace
+
+import (
+	"time"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+var _ = g.Describe("[Feature:Marketplace] Marketplace", func() {
+	defer g.GinkgoRecover()
+
+	var oc = exutil.NewCLI("marketplace", exutil.KubeConfigPath())
+
+	g.AfterEach(func() {
+		//clear the sub,csv resource
+		err := clearResources(oc, "subscription", "descheduler-test", "openshift-operators")
+		o.Expect(err).NotTo(o.HaveOccurred())
+		err = clearResources(oc, "csv", "descheduler.v0.0.3", "openshift-operators")
+		o.Expect(err).NotTo(o.HaveOccurred())
+		err = clearResources(oc, "catalogsourceconfig", "csctest", "openshift-marketplace")
+		o.Expect(err).NotTo(o.HaveOccurred())
+		err = clearResources(oc, "operatorsource", "opsrctest", "openshift-marketplace")
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+	})
+
+	//OCP-21405 create one opsrc,OCP-21419 create one csc,OCP-21479 sub to openshift-operators,OCP-21667 delete the csc
+	g.It("[ocp-21405 21419 21479 21667]create and delete the basic source,sub one operator", func() {
+		opsrcYaml := exutil.FixturePath("testdata", "marketplace", "opsrc", "01-opsrc.yaml")
+		err := createResources(oc, opsrcYaml)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		time.Sleep(10 * time.Second)
+		msg, _ := existResources(oc, "operatorsource", "opsrctest", "openshift-marketplace")
+		o.Expect(msg).Should(o.BeTrue())
+		msgofcat, _ := existResources(oc, "catalogsource", "opsrctest", "openshift-marketplace")
```

Comment:
For completeness I suggest checking for all the child resources like `Deployments` and `Services,` both here and for the `CatalogSourceConfig` test below. A stretch goal would be to also check if owner labels are getting set correctly.

Reply:
Thank you for the advice, I will add the test of other resources .This auto test will cover all the 
 OCP 4.1&4.2  now so do not contain the check of "owner labels"  this time since the difference of "owner labels" between 4.1 and 4.2.